src/Build/BackEnd/Components/BuildComponentFactoryCollection.cs

```diff
@@ -80,6 +82,8 @@ public void RegisterDefaultFactories()
 
             // SDK resolution
             _componentEntriesByType[BuildComponentType.SdkResolverService] = new BuildComponentEntry(BuildComponentType.SdkResolverService, MainNodeSdkResolverService.CreateComponent, CreationPattern.Singleton);
+
+            _componentEntriesByType[BuildComponentType.FileAccessManager] = new BuildComponentEntry(BuildComponentType.FileAccessManager, FileAccessManager.CreateComponent, CreationPattern.Singleton);
```

Comment:
Should this be under the ifdef? I am a bit surprised to see it everywhere but may not understand the layering you're thinking of.

Reply:
Yup, I was a little too conservative with the ifdefs to avoid sprinkling it everywhere. It wasn't too bad though so just went ahead and did it.